test/e2e/util/oauth.go

```diff
@@ -272,17 +272,30 @@ func validateClusterPreIDP(t *testing.T, ctx context.Context, client crclient.Cl
 func validateClusterPostIDP(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
 	g := NewWithT(t)
 
-	// update HC status
-	err := client.Get(ctx, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
-	g.Expect(err).ToNot(HaveOccurred())
-	g.Expect(hostedCluster.Status.KubeadminPassword).To(BeNil())
-
 	hcpNamespace := manifests.HostedControlPlaneNamespace(hostedCluster.Namespace, hostedCluster.Name).Name
 	kubeadminPasswordSecret := configmanifests.KubeadminPasswordSecret(hcpNamespace)
 	// validate kubeadmin secret is deleted
-	err = client.Get(ctx, crclient.ObjectKeyFromObject(kubeadminPasswordSecret), kubeadminPasswordSecret)
-	g.Expect(err).To(HaveOccurred())
-	g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+	err := wait.PollImmediateWithContext(ctx, 10*time.Second, 5*time.Minute, func(ctx context.Context) (bool, error) {
```

Comment:
why would we need 5min here?
shouldn't the status be updated right away as the removal of the idp triggers reconciliation?

Reply:
yes, status should be updated right away.  The wait here is for the secrete deletion.
Seth reported a flake in that the secret deletion check sometimes fail and suggested we add a wait here. The test hasn't flaked since then, so I think we can hold off on this one for now.